src/shared/boot-timestamps.c

```diff
@@ -33,12 +39,13 @@ int boot_timestamps(const dual_timestamp *n, dual_timestamp *firmware, dual_time
          * the monotonic timestamps here as negative of the actual
          * value. */
 
-        firmware->monotonic = y;
-        loader->monotonic = y - x;
-
-        a = n->monotonic + firmware->monotonic;
-        firmware->realtime = n->realtime > a ? n->realtime - a : 0;
+        if (discard_firmware) {
```

Comment:
Did you mean `!discard_firmware`?

Reply:
You're right. The perils of compile time testing. :D